compiler/x/amd64/codegen/OMRMemoryReference.hpp

```diff
@@ -67,7 +67,8 @@ class OMR_EXTENSIBLE MemoryReference : public OMR::X86::MemoryReference
    virtual uint8_t *generateBinaryEncoding(uint8_t *modRM, TR::Instruction  *containingInstruction, TR::CodeGenerator *cg);
 #endif
    TR::Register *getAddressRegister(){ return _addressRegister; }
-   void         setPreferRIPRelative(bool b) {_preferRIPRelative = b;}
+   void setForceRIPRelative(bool b) {_forceRIPRelative = b;}
```

Comment:
If you call `setForceRIPRelative(true)` on a memory reference that can't be expressed in RIP-relative terms (e.g. a MR with an index register) it generates a malformed instruction, where as `setPreferRIPRelative(true)` had no effect. Is that acceptable? If so maybe we should at least document this call to specify that the caller is responsible for making sure that the can be expressed in RIP-relative form?

Reply:
Agreed.  This API isn't a suggestion to use RIP-relative form, it's a requirement that this addressing mode be used.  That's why there's a fatal assert in binary encoding if the target is out of range.  I'll add a comment.

Reply:
Fixed in b3127535f69f5adf891b79179b19852d167b6fa9